appinventor/components/src/com/google/appinventor/components/runtime/Form.java

```diff
@@ -2632,6 +2632,19 @@ public void run() {
       });
   }
 
+  public void askPermissions(final String[] permissions) {
```

Comment:
There is currently a problem with the current `askPermission(BatchPermissionRequest ...)`; Probably not with old APIs, but with newer ones. The problem is that when requesting permissions in sequence, only the first one gets prompted, while all the others get dropped and treated as denied.

Hence, the standard way that would work is to use `ActivityCompat.requestPermissions` with an array argument in such cases. This was tested on API 26.

Maybe I'm missing something here but I have also encountered the problem before of those requests being dropped on API26

Reply:
Maybe I'm just being dense, but how are you handling the response here? You don't seem to be associating anything with the nonce to be handled when the request returns (asynchronously).

Reply:
Good point. It is not explicitly handled as such, but and remains implicitly handled by `ActivityCompat.requestPermissions((Activity)form, permissions, nonce);`;

As per docs:
> If your app does not have the requested permissions the user will be presented with UI for accepting them

I wasn't too sure if we need explicit handling for this case, because the options for the response are as follows:
1. The permissions are granted, and the app workflow continues as normal, with the permissions now being available as soon as they are accepted
2. The permissions are not granted, in which case the permission error falls through to the user (in the case of WebView, it will simply push a Toast notification with the error)

IIRC there was some buffer period for WebView to process the permissions, so as soon as they get accepted, they will be used instantly; So for the case of WebView it doesn't really make a difference. For other cases, I guess we can implement some callback system.

Reply:
But in my testing both dialogs appeared simultaneously. I could see the WebViewer dialog pop up before the App permission dialog, the latter just obscured the former. Can you do a version with the bulk permissions API call so that we can not show the dialog for the web viewer until the user has actually accepted permission? If they deny permission, that should also prevent the other dialog from showing...